src/main/shell-session/shell-session.ts

```diff
@@ -331,19 +336,27 @@ export abstract class ShellSession {
   }
 
   protected async getShellEnv() {
-    const shell = UserStore.getInstance().resolvedShell;
-    const env = clearKubeconfigEnvVars(JSON.parse(JSON.stringify(await shellEnv(shell || userInfo().shell))));
-    const pathStr = [await this.kubectlBinDirP, ...this.getPathEntries(), process.env.PATH].join(path.delimiter);
+    const shell = this.dependencies.resolvedShell || userInfo().shell;
+    const result = await this.dependencies.computeShellEnvironment(shell);
+    const rawEnv = (() => {
+      if (result.callWasSuccessful) {
+        return result.response ?? process.env;
+      }
+
+      return process.env;
+    })();
```

Comment:
Unlike in the `setupShellInjectable` case where we don't do anything if there are no new env vars to handle, here we need to continue.